### PR TITLE
VIH-7480 replace config resolve with config service

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7480

### Change description ###

Refactoring, replace config resolver with config service

**Does this PR introduce a breaking change?** (check one with "x")

```
[X ] Yes
[ ] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts [hunk 1]
```diff
@@ -86,7 +88,12 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
         this.loadJudges();
         this.initForm();
 
-        this.invalidPattern = this.route.snapshot.data['emailPattern'];
+        this.$subscriptions.push(
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:91]: You don't need to use the subscriptions observable - just do:

```ts
this.configService
                .getClientSettings()
                .pipe(map(x => x.test_username_stem))
                .subscribe(x => (this.invalidPattern = x));
```

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts [hunk 2]
```diff
@@ -58,13 +59,19 @@ export class SearchEmailComponent implements OnInit, OnDestroy {
...
 
     async getEmailPattern() {
-        const settings = await this.configService.getClientSettings().toPromise();
-        this.invalidPattern = settings.test_username_stem;
-        if (!this.invalidPattern || this.invalidPattern.length === 0) {
-            this.logger.error(`${this.loggerPrefix} Pattern to validate email is not set`, new Error('Email validation error'));
-        } else {
-            this.logger.info(`${this.loggerPrefix} Pattern to validate email is set with length ${this.invalidPattern.length}`);
-        }
+        this.$subscriptions.push(
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts:62]: As above, no need for subscriptions observable 👍

Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/search-email/search-email.component.ts]: Done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts (hunk 1 above)
Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts]: Done

File: AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts [hunk 3]
```diff
@@ -86,7 +88,10 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
         this.loadJudges();
         this.initForm();
 
-        this.invalidPattern = this.route.snapshot.data['emailPattern'];
+        this.configService
+            .getClientSettings()
+            .pipe(map(x => x.test_username_stem))
+            .subscribe(x => (this.invalidPattern = x));
```
Comment (phillipjchambers) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:94]: remove parens

Comment (valentinabennett) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:94]: Ahh... will do

Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts:94]: ignore this one

**Review by phillipjchambers — APPROVED** [review]

**Pull request merged**